core/aws-core/src/test/java/software/amazon/awssdk/awscore/exception/AwsServiceExceptionTest.java

```diff
@@ -82,6 +82,60 @@ public void exceptionMessage_withoutExtendedRequestId() {
         assertThat(e.getMessage()).isEqualTo("errorMessage (Service: serviceName, Status Code: 500, Request ID: requestId)");
     }
 
+    @Test
+    public void exceptionMessage_withAttempts() {
+        AwsServiceException e = AwsServiceException.builder()
+                                                   .message("errorMessage")
+                                                   .numAttempts(6)
+                                                   .awsErrorDetails(AwsErrorDetails.builder()
+                                                                                   .errorMessage("errorMessage")
+                                                                                   .serviceName("serviceName")
+                                                                                   .errorCode("errorCode")
+                                                                                   .build())
+                                                   .statusCode(500)
+                                                   .requestId("requestId")
+                                                   .build();
+
+        assertThat(e.getMessage()).contains("(SDK Diagnostics: numAttempts = 6)");
+        assertThat(e.numAttempts()).isEqualTo(6);
+    }
+
+    @Test
```

Comment:
Consider ParamterizedTest if we are just testing with different params.

Example: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/it/java/software/amazon/awssdk/services/s3/checksum/HttpChecksumIntegrationTest.java#L475